File: python/3718.smallest-missing-multiple-of-k.py
class Solution:
    def missingMultiple(self, nums: list[int], k: int) -> int:
        s = {n // k - 1 for n in nums if n % k == 0}

        for c in itertools.count():
            if c not in s:
                return (c + 1) * k

        return k

File: cpp/3718.smallest-missing-multiple-of-k.cpp
class Solution {
public:
    int missingMultiple(vector<int>& nums, int k) {
        bitset<101> s;

        for (int n : nums) {
          if (n % k == 0) {
            s.set(n / k - 1);
          }
        }

        s.flip();
        int pos = s._Find_first() + 1;

        return pos * k;
    }
};